ui/src/app/components/help-sidebar-icons.tsx

```diff
@@ -0,0 +1,667 @@
+            () => (
+              <FontAwesomeIcon
+                icon={faCircle}
+                style={{
+                  ...styles.asyncOperationStatusIcon,
+                  color: colors.asyncOperationStatus.succeeded,
+                }}
+              />
+            ),
+          ],
+          [
+            TerraJobStatus.ABORTED,
+            () => (
+              <FontAwesomeIcon
+                icon={faCircle}
+                style={{
+                  ...styles.asyncOperationStatusIcon,
+                  color: colors.asyncOperationStatus.stopped,
+                }}
+              />
+            ),
+          ]
+        )}
+      </FlexRow>
+    </FlexRow>
+  );
+};
+
+interface DisplayIconProps {
+  workspace: WorkspaceData;
+  criteria: Array<Selection>;
+  concept?: Array<Criteria>;
+  runtimeStore: RuntimeStore;
+  compoundRuntimeOps: CompoundRuntimeOpStore;
+  genomicExtractionJobs: GenomicExtractionJob[];
+}
+const displayIcon = (icon: IconConfig, props: DisplayIconProps) => {
+  const {
+    workspace,
+    runtimeStore,
+    compoundRuntimeOps,
+    genomicExtractionJobs,
+    criteria,
+    concept,
+  } = props;
+  return switchCase(
+    icon.id,
+    [
+      'dataDictionary',
+      () => (
+        <a href={supportUrls.dataDictionary} target='_blank'>
+          <FontAwesomeIcon
+            data-test-id={'help-sidebar-icon-' + icon.id}
+            icon={icon.faIcon}
+            style={icon.style}
+          />
+        </a>
+      ),
+    ],
+    [
+      'runtime',
+      () =>
+        displayRuntimeIcon(icon, workspace, runtimeStore, compoundRuntimeOps),
+    ],
+    [
+      'terminal',
+      () => (
+        <RouteLink
+          path={`/workspaces/${workspace.namespace}/${workspace.id}/terminals`}
+        >
+          <FontAwesomeIcon
+            data-test-id={'help-sidebar-icon-' + icon.id}
+            icon={icon.faIcon}
+            style={icon.style}
+          />
+        </RouteLink>
+      ),
+    ],
+    [
+      'genomicExtractions',
+      () =>
+        displayExtractionIcon(icon, genomicExtractionJobs, criteria, concept),
+    ],
+    [
+      DEFAULT,
+      () =>
+        icon.faIcon === null ? (
+          <img
+            data-test-id={'help-sidebar-icon-' + icon.id}
+            src={proIcons[icon.id]}
+            style={icon.style}
+          />
+        ) : (
+          displayFontAwesomeIcon(icon, criteria, concept)
+        ),
+    ]
+  );
+};
+
+export const showCriteriaIcon = (pageKey: string, criteria: Array<Selection>) =>
+  pageKey === 'cohortBuilder' && !!criteria;
+export const showConceptIcon = (pageKey: string) => pageKey === 'conceptSets';
+
+const runtimeTooltip = (baseTooltip: string, loadingError: Error): string => {
+  if (loadingError) {
+    if (loadingError instanceof ComputeSecuritySuspendedError) {
+      return `Security suspended: ${baseTooltip}`;
+    }
+    return `${baseTooltip} (unknown error)`;
+  }
+
+  return baseTooltip;
+};
+
+type IconKey =
```

Comment:
this is new - these strings are effectively used as enums, so we can add enforcement by making them a union-string type

Reply:
<img width="677" alt="Screen Shot 2022-10-28 at 5 08 35 PM" src="https://user-images.githubusercontent.com/2701406/198733245-e1c474e4-1270-4d73-a2ff-fad1bc0acb92.png">

Reply:
update: renamed to `SidebarIconId`